application-diagram-api/src/main/java/com/xwiki/diagram/internal/handlers/DiagramLinkHandler.java

```diff
@@ -176,48 +181,51 @@ public String getUserObjectNodeLink(String link)
      * 
-     * @param value link node in string format
-     * @return resource reference
+     * @param value text element content in string format
+     * @return resource references inside text element
      */
-    public String getMxCellNodeLink(String value)
+    public List<String> getMxCellNodeLinks(String value)
     {
+        List<String> links = new ArrayList<String>();
         if (value == null || value.indexOf(DiagramLinkHandler.HREF) == -1) {
-            return null;
+            return links;
         }
-
         try {
-            // The value attribute contains an 'a' node that holds the link value.
-            String link = getLinkFromEmbeddedNode(value);
-            if (link != null && isXWikiCustomLink(link)) {
-                return getResourceReferenceFromCustomLink(link);
-            }
+            // The value attribute contains the text element, which could contain one or more links.
+            links = getLinksFromEmbeddedNode(value);
+            links = links.stream().filter(link -> (link != null && isXWikiCustomLink(link)))
+                .map(link -> getResourceReferenceFromCustomLink(link)).collect(Collectors.toList());
         } catch (ParserConfigurationException | IOException | SAXException e) {
             logger.warn("Failed while parsing a mxCell node", e);
         }
 
-        return null;
+        return links;
     }
 
     /**
-     * Get link from value attribute of another node.
+     * Get links from the content of a text element.
      * 
-     * @param value the node that contains the link
-     * @return link inside node attribute
+     * @param value value of the text element.
+     * @return links inside text element
      * @throws ParserConfigurationException if document builder cannot be created
      * @throws IOException if parsing the document fails
      * @throws SAXException if parsing the document fails
      */
-    private String getLinkFromEmbeddedNode(String value) throws SAXException, IOException, ParserConfigurationException
+    private List<String> getLinksFromEmbeddedNode(String value)
+        throws SAXException, IOException, ParserConfigurationException
     {
-        // Create a DOM using the value and take the href attribute from inside the a element.
-        // TODO: iterate though children nodes since there could be multiple links inside the given value.
+        List<String> hrefValues = new ArrayList<String>();
+        // Create a DOM using value for getting the href of 'a' elements inside it.
         Document doc = defaultHTMLCleaner.clean(new StringReader(value));
-        Element node = (Element) doc.getElementsByTagName("a").item(0);
-        if (node != null) {
-            return node.getAttribute(HREF);
+        NodeList nodes = doc.getElementsByTagName("a");
+        for (int i = 0; i < nodes.getLength(); i++) {
+            Element node = (Element) nodes.item(i);
+            if (node != null) {
```

Comment:
Are you sure this can be null? It would be really strange if ``getElementsByTagName()`` would return null items.

Reply:
Indeed